HighwayPatrol.cpp

```diff
@@ -28,10 +35,30 @@ void HighwayPatrol::pullOver( Vehicle* v, bool willArrest, Highway* h )
     std::cout << name << ": vehicle is traveling " << v->speed - h->speedLimit << " miles per hour over the speed limit" << std::endl;
     if( willArrest )
     {
-        assert(false);
+        std::string vehicleType = "VEHICLE";
+        
+        if (Car* car = dynamic_cast<Car*>(v); car != nullptr)
+        {
+            vehicleType = "CAR";
+        }
+        else if (Motorcycle* motorcycle = dynamic_cast<Motorcycle*>(v); motorcycle != nullptr)
+        {
+            vehicleType = "MOTORCYCLE";
+        }
+        else if (SemiTruck* semiTruck = dynamic_cast<SemiTruck*>(v); semiTruck != nullptr)
+        {
+            vehicleType = "SEMITRUCK";
+        }
+        else if (HighwayPatrol* highwayPatrol = dynamic_cast<HighwayPatrol*>(v); highwayPatrol != nullptr)
+        {
+            vehicleType = "HIGHWAY PATROL";
+        }
```

Comment:
- use `auto*`.
- you don't need to explicitly check for `nullptr`.  

`nullptr` is effectively `0`, and the computer interprets `0` as `false`.  
if the `dynamic_cast` fails, the pointer will be `nullptr`, which evaluates in this context to `false`

- [ ] revise